### PR TITLE
add COSI support

### PR DESCRIPTION
**Pull request by giubacc**
add COSI support

Fixes: https://github.com/aquarist-labs/s3gw/issues/438
Fixes: https://github.com/aquarist-labs/s3gw/issues/436
# Describe your changes

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

**Review by m-ildefons — COMMENTED** [review]
A few small nits. None of them I'd consider blockers. One question does remain however: Do we want to provision a default bucket class?

**Comment by giubacc** [comment]
> A few small nits. None of them I'd consider blockers. One question does remain however: Do we want to provision a default bucket class?

Thx for the review.
I cannot say now if we should provision a default bucket class; probably we need to understand what is the best user experience here.
What if I create an issue to specifically handle this?

File: charts/s3gw/templates/cosi-driver-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            - secretRef:
+                name: {{ include "s3gw-cosi.driverSecretName" . }}
+          volumeMounts:
+            - mountPath: /var/lib/cosi
+              name: socket
+        - name: {{ .Release.Name }}-cosi-sidecar
+          image: {{ include "s3gw-cosi.sidecarImage" . | quote }}
+          imagePullPolicy: {{ default "IfNotPresent" .Values.cosi.sidecar.imagePullPolicy }}
+          args:
+            - {{ printf "--v=%s" (default "1" .Values.cosi.sidecar.logLevel) }}
```
[the remark this answers is not in the document]
Comment (giubacc) [reply charts/s3gw/templates/cosi-driver-deployment.yaml:40]: right, will fix.

**Review by m-ildefons — APPROVED** [review]
LGTM

**Pull request merged**